### ISSUE TITLE
Not working on iOS 11

### ISSUE THREAD

**Issue by dnish** [muaz-khan/RecordRTC#323]
When try to record on iOS 11, I get the following error log:

``` 
[Error] Missing analyticsSettings in Meteor.settings.public
	(anonymous function) (modules.js:53087)
	analyticsStartup (okgrow_analytics.js:418)
[Warning] URL.createObjectURL(stream) is deprecated, please use elem.srcObject = stream instead. (modules.js, line 53087)
[Log] started recording video stream. (modules.js, line 53087)
[Log] Using recorderType: – "WhammyRecorder" (modules.js, line 53087)
[Log] Using frames-interval: – 1 (modules.js, line 53087)
[Log] canvas resolutions – 336 – "*" – 448 (modules.js, line 53087)
[Log] video width/height – 336 – "*" – 448 (modules.js, line 53087)
[Log] Recorder state changed: – "recording" (modules.js, line 53087)
[Log] Initialized recorderType: – "WhammyRecorder" – "for output-type:" – "video" (modules.js, line 53087)
[Error] Unhandled Promise Rejection: [object DOMError]
	(anonymous function)
	rejectPromise
[Warning] Stopped recording video stream. (modules.js, line 53087)
[Log] Recorder state changed: – "stopped" (modules.js, line 53087)
[Error] InvalidCharacterError (DOM Exception 5): The string contains invalid characters. (blob:https://12a86d15.ngrok.io/8e98ba37-2dc0-4c74-861d-ee1feb8d71a3, line 317)
```

Also the video always opens in fullscreen, so the user can't press on stop.

**Comment by iamrototo** [comment]
Video not yet working, please refer to the browser compatibility:
https://github.com/muaz-khan/RecordRTC#browsers-support

**Comment by DrLongGhost** [comment]
I'd love to see audio **capture** implemented for Safari 11 for iOS and Desktop. I reached out to web-evangelist@apple.com and got the following reply. Posting it here in case it helps with this at all:

> 
> If you take another look at "A Closer Look Into WebRTC" (https://webkit.org/blog/7763/a-closer-look-into-webrtc/) you’ll note that it does not in fact say that Safari supports audio recording is supported. It states that capturing audio for streaming visa a MediaStream is supported.
> 
> Both the RecordRTC and Audio Recorder use MediaRecorder from the MediaStream Recorder API which Safari does not support. There is no support currently planned for MediaStream Recorder.
> 
> You can file a feature request for this API by filing a bug report with your use case for the API and the impact it would have to your user base. You can file bug reports at https://bugreport.apple.com
> 
> The WebAudio Input example fails because it calls the wrong API. It includes the following call:
> 
> 	var pc1 = new webkitRTCPeerConnection(servers); // eslint-disable-line new-cap
> 
> This fails because webkitRTCPeerConnection() is the wrong API call. It should be calling RTCPeerConnection() instead.
> 
> WebRTC and Media Capture is supported in Safari 11 and Safari Technology Preview on macOS Sierra and macOS High Sierra.
>

**Comment by john-landgrave** [comment]
@muaz-khan 

Given that iOS seems to support getUserMedia now, should this just start working, or is there more tinkering that needs to happen? This article appears to show a working demo: https://medium.com/@leemartin/hello-webrtc-on-safari-11-e8bcb5335295

**Comment by petrbela** [comment]
@john-landgrave iOS 11+ supports getUserMedia, MediaStream, and related APIs needed for WebRTC.

However, for local recording, there's a separate class called [MediaRecorder](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder). It's been available in Safari since Jan 2019 in preview but, as of today, it is still "Experimental" and users need to turn it on [in Safari settings](https://blog.addpipe.com/safari-technology-preview-73-adds-limited-mediastream-recorder-api-support/). It applies to both desktop Safari on Mac and mobile Safari on iOS.

Bottom line, unless you can tell your users to go change their browser settings, recording a video is not natively supported by Safari. RecordRTC has an alternative recorder called WhammyRecorder which circumvents the native APIs by hacking the stream through a canvas, but unfortunately, it doesn't currently work on Safari as mentioned in https://github.com/muaz-khan/RecordRTC/issues/380

**Comment by john-landgrave** [comment]
Thanks @petrbela , definitely appreciate the detailed response!
